### DISCUSSION TITLE
Rotating Rect

### DISCUSSION THREAD

**Discussion by Esaaj** [pymupdf/PyMuPDF#2112] [Looking for help] [answered]
I have the coordinates of a rectangle from a landscape pdf in which I am trying to insert an image or a checkbox, but it's
 ending up inserting in the wrong coordinates.

coordinates I got from
![Screenshot1](https://user-images.githubusercontent.com/48610882/206637667-bb31d26c-8e39-4fbd-8090-4996148deee7.jpg)

place it's getting inserted in.
![Screenshot2](https://user-images.githubusercontent.com/48610882/206637752-47bf858f-bd0a-49a0-85e4-720351e279b1.jpg)

**Comment by JorjMcKie** [comment, marked as answer]
Unfortunately, your images seem to cover some of your text. So I could not read all of it I think.
But some comment right away:
All coordinates from text extractions, line art, image bboxes, whatever are **_relative to the unrotated page_**.
There are `Page` attributes `.rotation_matrix` and `.derotation_matrix` to compute between the respective coordinate systems:
```python
pprint((page.transformation_matrix,page.rotation_matrix,page.derotation_matrix))
(Matrix(1.0, 0.0, 0.0, -1.0, 0.0, 842.0),
 Matrix(1.0, 0.0, 0.0, 1.0, 0.0, 0.0),
 Matrix(1.0, -0.0, -0.0, 1.0, 0.0, 0.0))
page.set_rotation(90)
pprint((page.transformation_matrix,page.rotation_matrix,page.derotation_matrix))
(Matrix(1.0, 0.0, 0.0, -1.0, 0.0, 842.0),
 Matrix(0.0, 1.0, -1.0, 0.0, 842.0, 0.0),
 Matrix(0.0, -1.0, 1.0, 0.0, -0.0, 842.0))
```

**Comment by Esaaj** [comment]
Thanks for the help. this helped me with the coordinates. Is there any way to rotate a checkbox after placing them?

**Comment by JorjMcKie** [comment]
> Is there any way to rotate a checkbox after placing them?

No, there is no such option.
